### ISSUE TITLE
`Format error: invalid uncompressed block len`

### ISSUE THREAD

**Issue by b52** [PistonDevelopers/image#425]
While trying to open the following image, I run into the an `FormatError` with the message `invalid uncompressed block len`. Any idea what might cause this and/or how to fix it, Image-wise and implementation wise?

![invalid image](https://paste.xinu.at/BZk/image.png)

The raw image can be found [here](https://paste.xinu.at/BZk/image.png).

**Comment by nwin** [comment]
That image works for me. Maybe github recompressed it?

**Comment by b52** [comment]
I updated my initial issue to provide the raw image.

**Comment by nwin** [comment]
That works as well on master, are you using master or the crates.io version?

**Comment by b52** [comment]
I built from the latest commit on `master`:

```
[[package]]
name = "image"
version = "0.3.11"
source = "git+https://github.com/PistonDevelopers/image#720dcc9d23058d66201baa8f7e2671c4fe21bb5e"
dependencies = [
 "byteorder 0.3.10 (registry+https://github.com/rust-lang/crates.io-index)",
 "enum_primitive 0.0.1 (registry+https://github.com/rust-lang/crates.io-index)",
 "gif 0.5.0 (registry+https://github.com/rust-lang/crates.io-index)",
 "glob 0.2.10 (registry+https://github.com/rust-lang/crates.io-index)",
 "num 0.1.25 (registry+https://github.com/rust-lang/crates.io-index)",
 "png 0.3.1 (registry+https://github.com/rust-lang/crates.io-index)",
]
```

**Comment by nwin** [comment]
Are you sure that the paste bin did not change the image? I opened that image with `cargo run --example opening <file>` and it worked fine.

**Comment by b52** [comment]
I had this error on several files, in total I opened about 400 images while 50 of those images caused the above described error. Interestingly, after I moved the images, which were located on a USB flash drive, to my HDD, the error was gone.
Not sure though why this might cause this specific error and not an `io.Error` or something similar. Therefore, I close this issue, but the error returned might be an incorrect one.
